llvm/include/llvm/Analysis/TargetTransformInfoImpl.h

```diff
@@ -427,6 +427,11 @@ class TargetTransformInfoImplBase {
     return TTI::TCC_Free;
   }
 
+  bool preferToKeepConstantsAttached(const Instruction &Inst,
+                                     const Function &Fn) const {
+    return true;
```

Comment:
Shouldn't this default implementation return false rather than true?

Reply:
Darn it. Thanks, I'll push an update.